test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSignCommandTest.cs

```diff
@@ -0,0 +1,523 @@
+            var mockConsole = new Mock<IConsole>();
+            var signCommand = new SignCommand
+            {
+                Console = mockConsole.Object,
+            };
+
+            signCommand.Arguments.Add(packagePath);
+
+            // Act & Assert
+            var ex = Assert.Throws<ArgumentException>(() => signCommand.GetSignArgs());
+            Assert.Equal(string.Format(_noArgException, nameof(SignCommand.Timestamper)), ex.Message);
+        }
+
+
+        [Fact]
+        public void SignCommandArgParsing_NoCertificate()
+        {
+            // Arrange
+            var packagePath = @"foo\bar.nupkg";
+            var timestamper = "http://foo.bar";
+            var mockSignCommandRunner = new Mock<ISignCommandRunner>();
+            var mockConsole = new Mock<IConsole>();
+            var signCommand = new SignCommand
+            {
+                Console = mockConsole.Object,
+                Timestamper = timestamper
+            };
+
+            signCommand.Arguments.Add(packagePath);
+
+            // Act & Assert
+            var ex = Assert.Throws<ArgumentException>(() => signCommand.GetSignArgs());
+            Assert.Equal(_noCertificateException, ex.Message);
+        }
+
+        [Theory]
+        [InlineData(@"\\foo\bar.pfx", "foo_bar_subject", "")]
+        [InlineData("\\foo\bar.cert", "", "foo_bar_fingerprint")]
+        [InlineData("\\foo\bar.cert", "foo_bar_subject", "foo_bar_fingerprint")]
+        [InlineData("", "foo_bar_subject", "foo_bar_fingerprint")]
+        public void SignCommandArgParsing_MultipleCertificateOptions(
+            string certificatePath,
+            string certificateSubjectName,
+            string certificateFingerprint)
+        {
+            // Arrange
+            var packagePath = @"foo\bar.nupkg";
+            var timestamper = "http://foo.bar";
+            var mockSignCommandRunner = new Mock<ISignCommandRunner>();
+            var mockConsole = new Mock<IConsole>();
+            var signCommand = new SignCommand
+            {
+                Console = mockConsole.Object,
+                Timestamper = timestamper,
+                CertificatePath =certificatePath,
+                CertificateSubjectName = certificateSubjectName,
+                CertificateFingerprint = certificateFingerprint
+            };
+
+            signCommand.Arguments.Add(packagePath);
+
+            // Act & Assert
+            var ex = Assert.Throws<ArgumentException>(() => signCommand.GetSignArgs());
+            Assert.Equal(_multipleCertificateException, ex.Message);
+        }
+
+        [Theory]
+        [InlineData("AddressBook")]
+        [InlineData("AuthRoot")]
+        [InlineData("CertificateAuthority")]
+        [InlineData("Disallowed")]
+        [InlineData("My")]
+        [InlineData("Root")]
+        [InlineData("TrustedPeople")]
+        [InlineData("TrustedPublisher")]
+        [InlineData("AddreSSBook")]
```

Comment:
I don't think you need to test the case-sensitivity of every `StoreName` member.  Having one case-insensitivity test is completely sufficient.